includes/class-controller.php

```diff
@@ -48,8 +50,46 @@ private function api_rewrite() {
 		}
 	}
 
+	/**
+	 * Ajax action to clear the Log file.
+	 *
+	 * @return void
+	 */
+	public function clear_log() {
+		if ( isset( $_POST['nonce'] ) && wp_verify_nonce( sanitize_key( $_POST['nonce'] ), 'aspireupdate-ajax' ) ) {
+			$status = Debug::clear();
+			wp_send_json_success(
+				[
+					'cleared' => $status,
+				]
+			);
+		}
+		wp_send_json_error();
+	}
+
+	/**
+	 * Ajax action to read the Log file.
+	 *
+	 * @return void
+	 */
+	public function read_log() {
+		if ( isset( $_POST['nonce'] ) && wp_verify_nonce( sanitize_key( $_POST['nonce'] ), 'aspireupdate-ajax' ) ) {
+			$content = Debug::read( 1000 );
+			if ( is_wp_error( $content ) ) {
+				$content = $content->get_error_message();
```

Comment:
```suggestion
				$content = $content->get_error_message();
       
```
Shouldn't we send a JSON error here instead?

Reply:
We are showing those messages inside the viewer itself, if we throw json error it will go into an alert box.

Reply:
On second thought I guess it make sense to throw those in the alert box. Let's reopen it